PLREQ/PLREQ/Models/PLREQDataManager.swift

```diff
@@ -68,29 +69,46 @@ class PLREQDataManager {
         return saveContext()
     }
     
+    // 플레이리스트 삭제
     func delete(playListObject: NSManagedObject) -> Bool {
         context.delete(playListObject)
         
         return saveContext()
     }
     
+    // 플레이리스트의 제목 변경
     func updateTitle(playListObject: NSManagedObject, title: String) -> Bool {
         playListObject.setValue(title, forKey: "title")
         
         return saveContext()
     }
     
+    // 플레이리스트의 음악 목록들을 불러오기
     func musicsFetch(playList: PlayListDB) -> [MusicDB] {
         return playList.music?.array as! [MusicDB]
     }
     
-    // 음악을 삭제하는 로직
+    // 음악을 삭제
     func musicDelete(music: NSManagedObject) -> Bool {
         context.delete(music)
         
         return saveContext()
     }
     
+    // 플레이리스트에 곡들을 추가
+    func addToPlayList(playListObject: NSManagedObject, musics: [Music]) -> Bool {
+        for music in musics {
+            let musicObject = NSEntityDescription.insertNewObject(forEntityName: MusicModelName, into: context) as! MusicDB
+            musicObject.title = music.title
+            musicObject.artist = music.artist
+            musicObject.musicImageURL = music.musicImageURL
+            
+            (playListObject as! PlayListDB).addToMusic(musicObject)
```

Comment:
안녕하세요 코인! 제가 코어데이터 오브젝트에 대해서 잘 알지 못해 혹시 playListObject와 PlayListDB의 관계에 대해서 설명해주실 수  있으실까요? xcode에서 fix하라는대로 playListObject를 playListDB로 강제 타입 캐스팅하곤 했었는데 이해를 하지 않고 사용했던 것 같아요! 키워드를 알려주셔도 괜찮습니다!

Reply:
CoreData는 Database가 아니라 객체를 나타내는 그래프 같은 느낌으로 `persistent`라는것을 이용해 데이터를 호출하고 저장하게 됩니다. `persistent`는 AppDelegate에 설명이 자세하게 나와있는데
```swift
/*
         The persistent container for the application. This implementation
         creates and returns a container, having loaded the store for the
         application to it. This property is optional since there are legitimate
         error conditions that could cause the creation of the store to fail.
        */
let container = NSPersistentContainer(name: "PLREQ")
```
요약하자면 `Core Data Stack을 나타내는 필요한 모든 객체를 포함한다`라고 할 수 있는데, 밑에 있는 `NSPersistentContainer`의 클래스를 보면
<img width="607" alt="image" src="https://user-images.githubusercontent.com/63584245/197478126-dd62f457-187c-43c1-bfc2-d3efa8ad71da.png">
저희가 CoreData를 사용하며 늘 쓰던 `NSManagedObjectContext`를 확인할 수 있습니다.

`NSManagedObjectContext`의 공식문서를 보면
<img width="792" alt="image" src="https://user-images.githubusercontent.com/63584245/197479121-a2de5f32-3a90-45b5-b791-8b720ca8083b.png">
`NSManagedObjectContext` 는 객체들의 변경사항을 관리한다 라고 되어 있습니다.

그래서 저희가 새로운 엔티티를 선언할 때 마다
```swift
let playListObject = NSEntityDescription.insertNewObject(forEntityName: playListModelName, into: NSManagedObjectContext)
```
처럼 `into: NSManagedObjectContext`를 사용해오고 있었던거였습니다.

그런데 `playListObject`는 `NSManagedObject`타입이기에 
```swift
extension PlayListDB {
    @NSManaged public func addToMusic(_ value: MusicDB)
}
```
`PlayListDB`에 선언된 함수인 `addToMusic`를 사용하지 못하기때문에 playListObject를 playListDB로 강제 캐스팅해오고 있었던 겁니다!